src/java/org/apache/cassandra/gms/FailureDetectorMBean.java

```diff
@@ -32,7 +32,9 @@
     public double getPhiConvictThreshold();
 
     @Deprecated public String getAllEndpointStates();
+    @Deprecated public String getAllEndpointStatesWithResolveIp();
```

Comment:
This smells. I do not know what is that method good for when we deprecated it on commit. Why do we introduce something we deprecate? Maybe I am missing something?

Reply:
@smiklosovic , I completely agree.
I have opted to use `@Deprecated` because the method `getAllEndpointStates` (which represents the same logic except for resolving ip) was marked as `@Deprecated`.
I couldn't understand myself why the method `getAllEndpointStates()` was marked as `@Deprecated` in the first place. The method is in use in fact and I have not found substitute logic for it in the code.

The [original commit](https://github.com/apache/cassandra/commit/59b5b6bef0fa76bf5740b688fcd4d9cf525760d0#diff-7957ef3fe425d75793e597cc24c9f1429c6da5aac9dd3fa6169ff832fec81814) by Ariel Weisberg however, introduced the method `@Deprecated getAllEndpointStates()` from start. It was never not deprecated.

I have followed related ticket [CASSANDRA-7544](https://issues.apache.org/jira/browse/CASSANDRA-7544) to find a clue for `@Deprecated` - but there was none.

For what it's worth, I suggest to remove `@Deprecated` for both of these methods.